File: packages/js-sdk/src/regex/detectors/uk/postcode.ts
// UK Postcode detector (scrubadub-inspired comprehensive pattern)

import { RegexDetector } from '../../base'
import { EntityType } from '../../entities'
import { registerRegion } from '../../registry'

class UkPostcodeDetector extends RegexDetector {
  entityType = EntityType.UK_POSTCODE
  score = 0.85
  // Comprehensive UK postcode pattern covering all valid formats
  pattern = new RegExp(
    '\\b(?:'
    + '[A-Z]{1,2}\\d[A-Z\\d]?'   // Area + district
    + '\\s?'
    + '\\d[A-Z]{2}'              // Sector + unit
    + ')\\b',
    'gi'
  )
}

registerRegion('uk', UkPostcodeDetector)

export { UkPostcodeDetector }

File: packages/python-sdk/src/blindfold/regex/detectors/uk/postcode.py
"""UK Postcode detector (scrubadub-inspired comprehensive pattern)"""

import re

from ...base import RegexDetector
from ...entities import EntityType
from ...registry import register_region


@register_region("uk")
class UkPostcodeDetector(RegexDetector):
    entity_type = EntityType.UK_POSTCODE
    score = 0.85
    # Comprehensive UK postcode pattern covering all valid formats
    pattern = re.compile(
        r"\b(?:"
        r"[A-Z]{1,2}\d[A-Z\d]?"  # Area + district
        r"\s?"
        r"\d[A-Z]{2}"            # Sector + unit
        r")\b",
        re.IGNORECASE,
    )
